app/controllers/api/v1/accounts/credentials_controller.rb

```diff
@@ -22,4 +23,20 @@ def update
   def account_params
     params.permit(:display_name, :note, :avatar, :header, :locked)
   end
+
+  def user_settings_params
+    return nil unless params.key?(:source)
+
+    source_params = params.require(:source).permit(
+      :privacy,
+      :sensitive
+    )
+    sensitive = source_params.key?(:sensitive) ? source_params[:sensitive] : object.user.setting_default_sensitive
+    ActionController::Parameters.new(
+      user: {
+        setting_default_privacy: source_params.key?(:privacy) ? source_params[:privacy] : object.user.setting_default_privacy,
+        setting_default_sensitive: sensitive ? '1' : '0', # database doesn't like boolean values for default_sensitive
+      }
+    ).require(:user).permit(:setting_default_privacy, :setting_default_sensitive).to_h
```

Comment:
Why not create a hash at first place if you eventually converts to hash?

Reply:
I tried just returning a hash that seemed identical to me, but the update failed when I did that, and I'm not sure why. I assumed there was some hidden side effect of doing it this way...

Reply:
...and there is - the key symbols are converted to strings. Updating...